### PR TITLE
Security module - being tested

### PR DESCRIPTION
**Pull request by jamescrowley321**

### CODE REVIEW

File: main.tf
Comment (hernanku) [inline main.tf:1]: Might be a good idea to make the `create_sns` into variables, and have a diff variable name for each of the module services 
`create_sns_topic  = true`  
`create_sns_topic  = var.create_security_hub_sns`

Comment (hernanku) [inline main.tf:1]: The cloudposse modules have a variable called `enabled` which controls module creating resources.
We could explore that in lieu of count?
image

File: example/main.tf [hunk 1]
```diff
@@ -2,19 +2,35 @@
...
 
+module "tags" {
+  source  = "sourcefuse/arc-tags/aws"
+  version = "1.2.3"
+
+  environment = var.environment
+  project     = var.project
+
+  extra_tags = {
+    Repo         = "github.com/sourcefuse/refarch-devops-infra"
```
[the remark this answers is not in the document]
Comment (vijay-stephen) [reply example/main.tf]: its done

File: sns.tf [hunk 2]
```diff
@@ -0,0 +1,67 @@
+locals {
+  kms_key_administrators = [
+    data.aws_iam_session_context.current.issuer_arn,
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+  ]
+
+  key_policy = jsonencode({
+    Id      = "key-policy-1",
```
Comment (tsaucier-sf) [inline sns.tf:8]: We should probably omit this if we can

File: example/data.tf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+data "aws_ssm_parameter" "guard_duty" {
+  name = "/${var.namespace}/poc/guard-duty/opsgenie/api-key"
```
Comment (tsaucier-sf) [inline example/data.tf:2]: change poc --> `${var.environment}`

File: example/data.tf [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+}
+
+data "aws_ssm_parameter" "security_hub" {
+  name = "/${var.namespace}/poc/security-hub/opsgenie/api-key"
```
Comment (tsaucier-sf) [inline example/data.tf:6]: change poc --> `${var.environment}`

File: example/data.tf [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+}
+
+data "aws_ssm_parameter" "aws_config" {
+  name = "/${var.namespace}/poc/aws-config/opsgenie/api-key"
```
Comment (tsaucier-sf) [inline example/data.tf:10]: change poc --> `${var.environment}`

**Review by tsaucier-sf — CHANGES_REQUESTED** [review]

File: main.tf [hunk 6]
```diff
@@ -13,87 +13,79 @@ terraform {
...
   version = "0.10.0"
 
+  count = var.enable_security_hub ? 1 : 0
+
   name = local.name_prefix
 
-  create_sns_topic  = var.create_sns_topic
-  enabled_standards = var.enabled_standards
-  subscribers       = length(var.security_hub_sns_subscribers) > 0 ? var.security_hub_sns_subscribers : local.security_hub_sns_subscribers
+  create_sns_topic  = true
```
[the remark this answers is not in the document]
Comment (vijay-stephen) [reply main.tf:24]: Problem is , `create_sns_topic` doesn't work for guard duty, Should I still create a variable ?

File: main.tf [hunk 7]
```diff
@@ -13,87 +13,79 @@ terraform {
...
+  count = var.enable_guard_duty ? 1 : 0
 
-  create_sns_topic      = var.create_sns_topic
-  enable_cloudwatch     = var.enable_cloudwatch
-  s3_protection_enabled = var.s3_protection_enabled
-  subscribers           = length(var.guard_duty_sns_subscribers) > 0 ? var.guard_duty_sns_subscribers : local.guard_duty_sns_subscribers
+  name = local.name_prefix
 
-  tags = module.tags.tags
+  create_sns_topic      = false
```
[the remark this answers is not in the document]
Comment (vijay-stephen) [reply main.tf]: Problem is , `create_sns_topic` doesn't work for guard duty, Should I still create a variable ?

File: sns.tf (hunk 2 above)
Comment (vijay-stephen) [reply sns.tf:8]: you mean remove the `Id` ?

File: main.tf
Comment (vijay-stephen) [reply main.tf:1]: create_sns_topic doesn't work for guard duty, thats why create_sns_topic is not exposed.

File: sns.tf (hunk 2 above)
Comment (tsaucier-sf) [reply sns.tf:8]: Yes

**Review by tsaucier-sf — APPROVED** [review]

File: main.tf
Comment (vijay-stephen) [reply main.tf:1]: Removed count and used `enabled`

**Pull request merged**